src/woo_search/search_index/api/viewsets/search.py

```diff
@@ -0,0 +1,109 @@
+from django.utils.translation import gettext_lazy as _
+
+from drf_spectacular.utils import extend_schema, extend_schema_view
+from rest_framework import viewsets
+from rest_framework.response import Response
+
+from woo_search.search_index.api.serializers import (
+    SearchResponseSerializer,
+    SearchSerializer,
+)
+
+from ...client import get_client
+from ...constants import SortChoices
+from ...typing import SearchType
+
+
+@extend_schema(tags=["index"])
+@extend_schema_view(
+    create=extend_schema(
+        summary=_("Search"),
+        description=_("Search the publication and document records."),
+        responses={
+            200: SearchResponseSerializer(many=True),
+        },
+    ),
+)
+class SearchViewSet(viewsets.ViewSet):
+    serializer_class = SearchSerializer
+    response_serializer_class = SearchResponseSerializer
+
+    def _query_multi_match(self, query: str):
+        return {
+            "multi_match": {
+                "fields": [
+                    "officiele_titel^2",
+                    "verkorte_titel^1.5",
+                    "omschrijving",
+                    "identifier^3",
+                ],
+                "fuzziness": 1,
+                "query": query,
+            }
+        }
+
+    def _build_must(self, params: SearchType):
+        must = []
+
+        if query := params.get("query"):
+            must.append(self._query_multi_match(query))
+
+        return must
+
+    def base_search_query(self, params: SearchType):
+        return {
+            "index": params.get("result_type"),
+            "size": params.get("page_size"),
+            "from": params.get("page"),
+            "track_total_hits": True,
+        }
+
+    def search_query(self, params: SearchType):
+        search = self.base_search_query(params)
+
+        if sort := params.get("sort"):
+            if sort == SortChoices.chronological:
```

Comment:
Misschien even checken bij Marco wat de verschillende sorteringen moeten doen.
Hoe ik het in mn hoofd had was:
```py
// relevance
[{"_score": {"order": "desc"}}, {"laatst_gewijzigd_datum": {"order": "desc"}}]

// chronological
[{"laatst_gewijzigd_datum": {"order": "desc"}}, {"_score": {"order": "desc"}}]
```